src/tube/watcher.js

```diff
@@ -69,10 +69,16 @@ export default class Watcher {
       if (err.message === 'DEADLINE_SOON') {
         await Promise.delay(500);
       }
+      if (err.message.includes('ECONNREFUSED')) {
+        if (onConnectionError) {
+          onConnectionError(err);
```

Comment:
Maybe `onConnectionError` should be able to return a bool that if `false` will stop the loop completely?

Reply:
I added an example in the README (and in the test) how to stop the loop.